lib/scope.js

```diff
@@ -1805,34 +1805,35 @@ module.exports = {
     /* Try to upload one or more files to a file-type input field.
     *    Does not submit, so no danger of timeout with server reload.
     */
-    
-    // Tests could be in one of two places. Figure out which directory they're in
-    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
-    let dir_source = `${ process.cwd() }/docassemble/${ session_vars.get_repo_url().match(/docassemble-(.*)/)[1] }/data/sources`;
-    let dir_features = `${ process.cwd() }/tests/features`;
-    let dir = null;
 
-    // Make sure one of the valid directories exist. This only allows
-    // one directory to be used per Step or table row. We don't anticipate
-    // future people having a `tests/features` folder, let alone both.
+    // Tests should be one place: docassemble/*/data/sources, where * is the only directory in `docassemble`.
+    // Makes an absolute path from the directory in which `npm run` was used plus our internal folders
+    let docassemble_pkg_path = `${ process.cwd() }/docassemble`;
+    let pkg_path = null;
     try {
-      // Try the most common directory first
-      fs.accessSync( dir_source );
-      dir = dir_source;
-      // If they have both directories, let them know which directory will be used.
-      try {
-        fs.accessSync( dir_features );
-        await scope.addToReport( scope, { type: `warning`, value: `You have both a docassemble package 'sources' folder and a 'test/features' folder. These tests will use the package's 'sources' folder.` });
-      } catch ( error_features ) { `all's well, do nothing`; }
-
-    } catch ( error_source ) {
-      try {
-        fs.accessSync( dir_features );
-        dir = dir_features;
-      } catch ( error_features ) {
-        await scope.addToReport( scope, { type: `warning`, value: `There are two places where you can store files to upload: "${ dir_source }" and "${ dir_features }". These tests could not find either of those.` });
-        throw error_source;  // Most people will want to know the sources folder path;
+      fs.accessSync(docassemble_pkg_path);
+      // https://stackoverflow.com/a/24594123/11416267
+      let dirs = fs.readdirSync(docassemble_pkg_path, {withFileTypes: true}).filter(dirent => dirent.isDirectory()).map(dirent => dirent.name);
+      if (dirs.length < 1) {
+        let msg = `You can store files to upload in 'docassemble/*/data/source'. These tests could not find a directory like that.`;
```

Comment:
Nitpick: Can we include the package name in `'docassemble/*/data/source'` instead of `*`? I think at this point we already have it. For less experienced developers, the `*` would be confusing.

Reply:
Do you mean from the repo name? It's an okay guess, but TBH if they don't have a `docassemble/*` directory, it's not a docassemble package (you can't put things in top level docassemble namespace), and there are other issues that we won't be able to help with? Essentially, this should be a very rarely seen error.

Reply:
Added it, wasn't able to improve the error message much more in this case, but if they don't have a docassemble folder I do ask if they're in the right directory (printing the current working directory as well).